### DISCUSSION TITLE
Next version of PCBs (please upgrade your firmware)

### DISCUSSION THREAD

**Discussion by revk** [revk/xESP32-Faikout#60] [Announcements]
I'm working on the next batch of PCBs.

If you have one of my [#Faikin](https://toot.me.uk/tags/Faikin) modules for control of Daikin aircon (https://www.amazon.co.uk/dp/B0C2ZYXNYQ) then I recommend you upgrade the software.

Got to web control page, Settings, Upgrade, simple as that.

Important features are.

1. Automatic upgrade (can be disabled by MQTT based setting otaauto setting to 0)

2. "Locking" the GPIO config used on your board in flash, so later changes to defaults in code will not change your config.

Yes, there is a planned change to one of the GPIO settings in future.

The next batch of boards are expected to have:

- New regulator
- New ESP programming header pads, staggered to hold the pins better (mainly for my benefit)
- New LED as JLCPCB had issues with the one we were using
- Swapped the connector to the other side so it can "protect" the various components by its height when posted by Amazon.
- Changed to 0402 passives
- New panel and silk screen making neater edges and easier to snap out
- Even new "tear drop" feature in #KiCad
- Building as 1.0mm not 1.6mm so easier to snap off
- Oh, and probably "black" because, why not?!
- FET in and out to allow 5V working (was just input before)

So lots of minor tweaks really.

But looking good. All worked the same, same size. Will have prototypes of these soon to test.

![a](https://github.com/revk/ESP32-Faikin/assets/996983/d00a9f86-140f-4ee6-903d-4ec4842b000e)
![b](https://github.com/revk/ESP32-Faikin/assets/996983/48636856-5b07-493e-a074-71acc61731d9)

Feedback on any other changes welcome...

Update: Further discussion with a customer has found that some Daikin do not work because the Faikin drives the Tx line with 3.3V signalling. A solution has been devices that uses open collector against the 5V input on pin 1. But as pin 1 is not always present, it is hoped this will work anyway as the Tx pin seems to have some internal pull up as well. More testing will be done on this change next week.

![image](https://github.com/revk/ESP32-Faikin/assets/996983/b0acb9ca-8348-48ac-a73a-ada5b0e64c4b)

**Comment by njh** [comment]
Exciting! It is amazing to see that there is demand for what is seemingly a very niche product.

- Are there any benefits to the new Regulator?
- Did you include a protection diode in this version?
- What would happen if you connected the header the wrong way round?
  - I guess you would have 14.2V to GPIO 27? But maybe no ground connected...

**Reply by revk** [reply]
OK, as you see, I have added BAV99S on the Tx/Rx at the CPU, they also have 100Ω and a cap. And removed now FETs in front.

As for reverse connector, the GND would go to NC, and the DC would go to the base of a FET on the Rx side, so even that would not have a useful potential against the Tx line (only other connection). So I think all pretty safe.

The new regulator is cheaper, and avoids the need of a diode (which JLC seem fond of fitting backwards some times) as it is synchronous.

But good points, thanks.

**Comment by ddv2005** [comment]
Keep in mind that the Faikin3 design using the 1st pin as VCC for TX but on the FTXR series this pin is not connected and I guess that new design will not work.

**Reply by revk** [reply]
I have tested on my Daikin here which as now pin 1, it seems there is an internal pull up to 5V.

Can you check voltage on you FTXR Rx pin when not connected, is it pulled up to 5V?

**Reply by ddv2005** [reply]
There are no wire on connector. 
![image](https://github.com/revk/ESP32-Faikin/assets/1864211/e4ce0bc0-922d-4ec0-863a-19d1b65c48ca)

**Reply by revk** [reply]
There is an Rx wire on the connector, is that 5V (i.e. pulled up), if so, then it should work.
